### ISSUE TITLE
Doesn't parse special-case TLD's

### ISSUE THREAD

**Issue by reggi** [wankdanker/node-domain-name-parser#1]
http://stackoverflow.com/questions/569137/how-to-get-domain-name-from-url

``` bash
  www.businessinsider.com.au // hostname
  com.au // domainName
  com // domain
```

**Comment by wankdanker** [comment]
Good point. I didn't read all of the stack over flow question, so maybe this question is answered there...

What would you expect the domain name and domain to be in this case? My assumption would be

```
www.businessinsider.com.au // hostname
businessinsider.com.au // domainName
businessinsider.com // domain
or
businessinsider //domain
```

I don't deal with these types of TLDs in my daily work so have not had to address this issue nor do I have expertise when it comes to what is expected. 

Dan

**Comment by reggi** [comment]
I'd expect

```
www.businessinsider.com.au // hostname
businessinsider.com.au // domainName
businessinsider // domain
```

**Comment by reggi** [comment]
It's understandable that you don't work with those, though! Feel free to close the issue!

**Comment by reggi** [comment]
The ideal domain name parsing library would have a data store of all viable TLDs I guess.

**Comment by wankdanker** [comment]
I'm all about making this module correct. :) I was trying to do it in a way that did not require awareness of legitimate TLDs, though.

Although, this specific case may be just related to TLDs that are a country's TLD. I would not have a problem including code specific to handling country-code TLDs because there (I think) is a finite number of countries in the world. :) Or at least it would change very rarely.

It would be pretty trivial to use a module like https://www.npmjs.com/package/countries-list and detect if the TLD is in the country list and then do special processing to return the results you expect.

I'll keep this open until this happens.

Thank you for pointing this out.

**Comment by wankdanker** [comment]
@reggi, This should be fixed in [v0.0.6](https://github.com/wankdanker/node-domain-name-parser/tree/v0.0.6).

Thank you.

**Comment by wankdanker** [comment]
FYI: @reggi, @tbuchok in 2.0.0 this behavior is behind an options flag called `mergeCountry`. If your package deps are locked in the 1.x.x series then you shouldn't run into any issues. See the docs for more info; you can set a global default.
